book/content/temperatures/temperatures_basic.md

```diff
@@ -0,0 +1,211 @@
+  language: python
+  name: python3
+---
+
+# Basic functionality #
+
+Definition of a temperature field or problem is essential for hydrogen transport and FESTIM as a whole. Users can define it as a fixed value or as a function of space and time. Read more about temperature in FESTIM _[here](https://festim.readthedocs.io/en/fenicsx/userguide/temperature.html)_. FESTIM's unit for temperature is Kelvin.
+
+This tutorial will go over the basics of defining the temperature in a FESTIM simulation.
+
+Objectives:
+* Learn how to define homogeneous temperature
+* Learn how to define space and time dependent temperature functions
+* Learn how to define temperature as a `dolfinx.fem.Function`
+
++++
+
+## Defining temperatures in FESTIM ##
+
+The simplest way to define temperature is as a homogeneous value:
+
+```{code-cell} ipython3
+import festim as F
+
+my_model = F.HydrogenTransportProblem()
+my_model.temperature = 300  # K 
+```
+
+To define a space-dependent temperature function we can use ` lambda ` functions:
+
+$$ 
+
+T(x) = T_0 e^{-x}
+
+$$
+
+```{code-cell} ipython3
+import festim as F
+import ufl
+
+T0 = 300
+my_model = F.HydrogenTransportProblem()
+my_model.temperature = lambda x: T0 * ufl.exp(-x)  # K
+```
+
+For a time-dependent function such as:
+
+$$
+
+T(t) = T_0 sin(t)
+
+$$
+
+```{code-cell} ipython3
+my_model.temperature = lambda t: T0 * ufl.sin(t)
+```
+
+For a function in $ x, y, z, t $ space:
+
+$$
+
+T(x,y,z,t) = T_0 cos(x)sin(t) + 5y - 2z
+
+$$
+
+```{code-cell} ipython3
+my_model.temperature = lambda x,t: T0 * ufl.cos(x[0])*ufl.sin(t) + 5*x[1] - 2*x[2]
+```
+
+<div class="alert alert-block alert-info">
+When defining custom functions for values, only the arguments x, t and T can be defined. Spatial coordinates can be referred to by their indices, such as x[0], x[1], and x[2], regardless of the coordinate system used. Time dependence must use t, and T for temperature dependence.
+</div>
+
++++
+
+We can assign a unit cube mesh and run the simulation to see how a spatially-dependent temperature field affects the transport:
+
+```{code-cell} ipython3
+from dolfinx.mesh import create_unit_cube
+from mpi4py import MPI
+import ufl
+import festim as F
+import numpy as np
+
+mesh = F.Mesh(create_unit_cube(MPI.COMM_WORLD, 10, 10, 10))
+my_model.mesh = mesh
+
+mat = F.Material(D_0=1, E_D=0)
+
+volume = F.VolumeSubdomain(id=1, material=mat)
+top_surface = F.SurfaceSubdomain(id=1, locator=lambda x: np.isclose(x[2], 1.0))
+bottom_surface = F.SurfaceSubdomain(id=2, locator=lambda x: np.isclose(x[2], 0.0))
+my_model.subdomains = [top_surface, bottom_surface, volume]
+
+H = F.Species("H")
+my_model.species = [H]
+
+my_model.boundary_conditions = [
+    F.FixedConcentrationBC(subdomain=top_surface, value=1.0, species=H),
+    F.FixedConcentrationBC(subdomain=bottom_surface, value=0.0, species=H),
+]
+
+my_model.settings = F.Settings(atol=1e-10, rtol=1e-10, transient=False)
+
+my_model.initialise()
+my_model.run()
+```
```

Comment:
We could add the temperature field in this example to see it in action. Maybe create a new model object too with a different name to distinguish it